zookeeper-client/zookeeper-client-c/src/st_adaptor.c

```diff
@@ -77,11 +77,11 @@ int32_t inc_ref_counter(zhandle_t* zh,int i)
 
 int32_t get_xid()
 {
-    static int32_t xid = -1;
-    if (xid == -1) {
-        xid = time(0);
-    }
-    return xid++;
+    static int32_t xid = 1;
+
+    // The XID returned should not be negative to avoid collisions
+    // with reserved XIDs, such as AUTH_XID or SET_WATCHES_XID.
+    return xid++ & ~(1<<31);
```

Comment:
I realize this will sound very boring but why not just reset xid to 1 prior to it rolling over? a) the xid actually represents the next xid to send and b) much easier to read/grok for someone new to the code.

Reply:
I'm not a C expert, but I believe C coders generally prefer magic bitwise operations to gain performance.

Reply:
The obscurity over-weights the benefits to me... (a/b earlier)

Reply:
Sorry for the delay. I just updated the patch such that for the single threaded version of get_xid we now reset the xid counter to 1 if a previous increment has caused an overflow.